### PR TITLE
Update waitForPin to use readMVar instead of take

### PR DESCRIPTION
**Pull request by nhibberd**
! @markhibberd @charleso 

Idea here is that we want to be able to have multiple things waiting for a pin and to have them all succeed as opposed to only having one of them succeed. 
/jury approved @charleso

### CODE REVIEW

**Comment by charleso** [comment]
👍

File: test/Test/Twine/Data/Pin.hs [hunk 1]
```diff
@@ -52,6 +54,18 @@ prop_pull_pull = once . testIO $
...
+
+--
+-- Multiple things can be blocked on a pin
+--
+prop_multiple = once . testIO $
+  newPin >>= \p -> do
+   x <- async $ waitForPin p
+   y <- async $ waitForPin p
+   pullPin p
+   r <- waitBoth x y
```
Comment (tmcgilchrist) [inline test/Test/Twine/Data/Pin.hs]: Is it valid to also do the waits in sequence?

```haskell
 pullPin p
r1 <- wait x
r2 <- wait y
```

Comment (nhibberd) [inline test/Test/Twine/Data/Pin.hs]: Yeah it is, will add a test for it

Comment (tmcgilchrist) [inline test/Test/Twine/Data/Pin.hs]: 👍

**Pull request merged**